sdk/communication/AzureCommunicationChat/Source/Signaling/CommunicationSignalingClient.swift

```diff
@@ -149,8 +149,38 @@ class CommunicationHandler: NSObject, TrouterListener {
             let chatEventId = try ChatEventId(forCode: generalPayload._eventId)
 
             // Convert trouter payload to chat event payload
-            let chatEvent = try TrouterEventUtil.create(chatEvent: chatEventId, from: request)
-            handler(chatEvent, chatEventId)
+            switch chatEventId {
+            case ChatEventId.chatMessageReceived:
+                let chatMessageReceivedEvent = try ChatMessageReceivedEvent(from: request)
+                handler(chatMessageReceivedEvent, chatEventId)
```

Comment:
Effectively, this doesn't get rid of `Any` because the typealias `EventHandler` specifies type `Any`.